versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java

```diff
@@ -932,32 +923,45 @@ protected abstract void doWriteRefLog(NonTransactionalOperationContext ctx, RefL
   protected abstract void unsafeWriteGlobalPointer(
       NonTransactionalOperationContext ctx, GlobalStatePointer pointer);
 
-  protected Hash writeGlobalCommit(
+  protected GlobalStateLogEntry writeGlobalCommit(
       NonTransactionalOperationContext ctx,
       long timeInMicros,
-      Hash parentHash,
+      GlobalStatePointer pointer,
       List<ContentIdAndBytes> globals)
       throws ReferenceConflictException {
-    GlobalStateLogEntry currentEntry = fetchFromGlobalLog(ctx, parentHash);
 
-    Stream<Hash> newParents = Stream.of(parentHash);
-    if (currentEntry != null) {
+    Hash parentHash = Hash.of(pointer.getGlobalId());
+    Hash hash = randomHash();
+
+    Stream<ByteString> newParents;
```

Comment:
can you elaborate more on this logic?  (Maybe ad a Javadoc comment)

Reply:
added some inline docs